File: Editor/src/main/java/org/turbomedia/turboedit/editor/renderer/RenderServerEntry.java
package org.turbomedia.turboedit.editor.renderer;

import org.turbomedia.turboedit.editor.misc.PreferencesFile;

public final class RenderServerEntry {
    private String displayName;
    private String ip;
    private FileResolveMethod fileResolveMethod;
    private boolean defaultServer;
    private final boolean buildIn;

    public RenderServerEntry(String displayName, String ip, FileResolveMethod fileResolveMethod, boolean defaultServer, boolean buildIn) {
        this.displayName = displayName;
        this.ip = ip;
        this.fileResolveMethod = fileResolveMethod;
        this.defaultServer = defaultServer;
        this.buildIn = buildIn;
    }

    @Override
    public String toString() {
        return displayName + (PreferencesFile.CURRENT_PREFERENCES.showIPsForServers ? " [" + ip + "]" : "") + (buildIn ? " [BUILD-IN]" : "") + (defaultServer ? " [DEFAULT]" : "");
    }

    public String displayName() {
        return displayName;
    }

    public void displayName(String displayName) {
        this.displayName = displayName;
    }

    public String ip() {
        return ip;
    }

    public void ip(String ip) {
        this.ip = ip;
    }

    public FileResolveMethod fileResolveMethod() {
        return fileResolveMethod;
    }

    public void fileResolveMethod(FileResolveMethod fileResolveMethod) {
        this.fileResolveMethod = fileResolveMethod;
    }

    public boolean defaultServer() {
        return defaultServer;
    }

    public void defaultServer(boolean defaultServer) {
        this.defaultServer = defaultServer;
    }

    public boolean buildIn() {
        return buildIn;
    }
}

File: Editor/src/main/kotlin/turboedit/editor/renderer/RenderServerEntry.kt
package turboedit.editor.renderer

import turboedit.editor.misc.Locale
import turboedit.editor.misc.PreferencesFile

class RenderServerEntry(
    private val id: String?,
    private var displayName: String?,
    private var ip: String,
    private var fileResolveMethod: FileResolveMethod?,
    private var defaultServer: Boolean,
    private val buildIn: Boolean
) {
    override fun toString(): String {
        return displayName + (if (PreferencesFile.CURRENT_PREFERENCES!!.showIPsForServers) " [$ip]" else "") + (if (buildIn) " [" + Locale.getText(
            "render_server_entry.build_in"
        ) + "]" else "") + (if (defaultServer) " [" + Locale.getText("render_server_entry.default") + "]" else "")
    }

    fun id(): String? {
        return id
    }

    fun displayName(): String? {
        return displayName
    }

    fun displayName(displayName: String?) {
        this.displayName = displayName
    }

    fun ip(): String {
        return ip
    }

    fun ip(ip: String) {
        this.ip = ip
    }

    fun fileResolveMethod(): FileResolveMethod? {
        return fileResolveMethod
    }

    fun fileResolveMethod(fileResolveMethod: FileResolveMethod?) {
        this.fileResolveMethod = fileResolveMethod
    }

    fun defaultServer(): Boolean {
        return defaultServer
    }

    fun defaultServer(defaultServer: Boolean) {
        this.defaultServer = defaultServer
    }

    fun buildIn(): Boolean {
        return buildIn
    }
}